mlflow/environment_variables.py

```diff
@@ -34,6 +38,24 @@ def __repr__(self):
         return repr(self.name)
 
 
+class _BooleanEnvironmentVariable(_EnvironmentVariable):
```

Comment:
For this class, we can define a new `__init__` ctor, like
```
    def __init__(self, name, default):
        self.name = name
        self.type = bool
        self.default = default
        assert default in [True, False]
```

Reply:
updated the constructor to perform validation on the default value.